### PR TITLE
DM-39857: Stop using pytest-flake8 and add ruff configuration

### PR DESCRIPTION
**Pull request by timj**
## Checklist

- [ ] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

**Comment by timj** [comment]
@zhaoyuyoung these are lots of small cleanups.

File: python/lsst/ctrl/bps/panda/cli/panda_auth.py
Comment (zhaoyuyoung) [inline python/lsst/ctrl/bps/panda/cli/panda_auth.py:1]: Looks good.

File: python/lsst/ctrl/bps/panda/cmd_line_embedder.py
Comment (zhaoyuyoung) [inline python/lsst/ctrl/bps/panda/cmd_line_embedder.py:1]: Looks good.

File: python/lsst/ctrl/bps/panda/edgenode/cmd_line_decoder.py
Comment (zhaoyuyoung) [inline python/lsst/ctrl/bps/panda/edgenode/cmd_line_decoder.py:1]: Looks good.

File: python/lsst/ctrl/bps/panda/panda_service.py
Comment (zhaoyuyoung) [inline python/lsst/ctrl/bps/panda/panda_service.py:1]: Looks good.

File: python/lsst/ctrl/bps/panda/utils.py
Comment (zhaoyuyoung) [inline python/lsst/ctrl/bps/panda/utils.py:1]: Looks good.

File: tests/test_panda_auth_utils.py
Comment (zhaoyuyoung) [inline tests/test_panda_auth_utils.py:1]: Looks good.

**Comment by zhaoyuyoung** [comment]
Hi Tim,
I've checked all the changes in the python codes/comments. They all look good to me.  I was stumbled by your change in https://github.com/lsst/ctrl_bps_panda/blob/main/python/lsst/ctrl/bps/panda/utils.py#L157 to
     if not isinstance(ret, list | tuple)
but figured out it's fine with python 3.10

For pytest-flake8 and ruff, I don't have much to say. Just wonder does the failure of the required "build_and_test / build_and_test (3.10) (pull_request) " matter?

**Comment by timj** [comment]
The breakage in the build is due to the pydantic change and upstream packages on PyPI don't yet work with pydantic 2 and have not pinned. I can either change requirements.txt temporarily to look at branches for daf_butler or change it to pin pydantic or else wait to merge until tomorrow when the PyPI gets versions that support pydantic 2.

**Comment by zhaoyuyoung** [comment]
Either way is ok. Curious, so the upstream daf_butler in requirements.txt is pinning pydantic 2?

**Comment by timj** [comment]
No, upstream hadn't pinned pydantic at all and so v2 is being used here. I have fixed `main` so that pydantic 2 works but this action is loading daf_butler from PyPI and so is getting the one that does not work with v2. Tonight we will update pypi and this PR will work fine.

**Review by zhaoyuyoung — APPROVED** [review]
As discussed there are no code except some style changes in this branch. A tiny testing workflow was submitted to PanDA and ran successfully. I think it's safe to merge this PR when you think the unit test works with pydantic2.

**Pull request merged**